src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -10,16 +10,40 @@ namespace System.Formats.Tar
     /// Even though the <see cref="TarEntryFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in that POSIX standard.</remarks>
     public abstract partial class PosixTarEntry : TarEntry
     {
-        // Constructor used when reading an existing archive.
-        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin)
-            : base(header, readerOfOrigin)
+        // Constructor called when reading a TarEntry from a TarReader.
+        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin, TarEntryFormat format)
+            : base(header, readerOfOrigin, format)
         {
         }
 
-        // Constructor called when creating a new 'TarEntry*' instance that can be passed to a TarWriter.
         internal PosixTarEntry(TarEntryType entryType, string entryName, TarEntryFormat format)
             : base(entryType, entryName, format)
         {
+            _header._uName = string.Empty;
+            _header._gName = string.Empty;
+            _header._devMajor = 0;
+            _header._devMinor = 0;
+        }
+
+        // Constructor called when creating a new TarEntry.
+        internal PosixTarEntry(TarEntry other, TarEntryFormat format)
+            : base(other, format)
+        {
+            // Avoid transferring if entry was V7
+            if (other is PosixTarEntry)
+            {
+                _header._uName = other._header._uName;
+                _header._gName = other._header._gName;
+                _header._devMajor = other._header._devMajor;
+                _header._devMinor = other._header._devMinor;
+            }
+            else
+            {
+                _header._uName = string.Empty;
+                _header._gName = string.Empty;
+                _header._devMajor = 0;
+                _header._devMinor = 0;
+            }
```

Comment:
Is this `else` needed?  If it was deleted, what value would these fields have?  (Maybe the string.Empty assignments are needed, but the devMajor/devMinor assign-0 seems redundant with being in the constructor)

Reply:
Not needed. The only thing I do need is ensuring uname and gname are never null.